### ISSUE TITLE
Harcoded WP_CONTENT_DIR wp-content not dynamic

### ISSUE THREAD

**Issue by cconversion** [publishpress/PublishPress#317]
Offending harcoded string - make dynamic - **_'wp-content/plugins/'_**

publishpress\vendor\alledia\wordpress-plugin-framework\src\library\Container.php#L104
```
  102                  }
  103  
  104:                 $data = get_plugin_data(ABSPATH . '/wp-content/plugins/' . $c['PLUGIN_BASENAME']);
  105  
  106                  return $data['Name'];
```

publishpress\vendor\alledia\wordpress-plugin-framework\src\library\Module\Addons.php#L483
```
  481  
  482          foreach ($pool as $addon) {
  483:             if (file_exists(ABSPATH . 'wp-content/plugins/' . $addon['slug'] . '/' . $addon['slug'] . '.php')) {
  484                  add_action('plugin_action_links_' . $addon['slug'] . '/' . $addon['slug'] . '.php',
  485                      [$this, 'actionSetLicenseKeyLink'], 998, 2);
```

### Error Console:

Warning | fopen(/home/hostingusername/public_html/subdirsitename//wp-content/plugins/publishpress/publishpress.php):   failed to open stream: No such file or directory
-- | --

-
1.	do_action('init')
wp-settings.php:467
2.	PublishPress()
wp-includes/class-wp-hook.php:286
3.	PublishPress\N\Plugin->__construct()
io/plugins/publishpress/publishpress.php:953
4.	PublishPress\N\Plugin->get_service()
io/plugins/publishpress/libraries/Notifications/Plugin.php:27
5.	Pimple\Container->offsetGet()
io/plugins/publishpress/libraries/Notifications/Traits/Dependency_Injector.php:41
6.	PublishPress\N\Pimple_Container::PublishPress\N\{closure}()
io/plugins/publishpress/vendor/pimple/pimple/src/Pimple/Container.php:118
7.	Allex\Core->__construct()
io/plugins/publishpress/libraries/Notifications/Pimple_Container.php:106
8.	Allex\Core->get_service()
io/plugins/publishpress/vendor/alledia/wordpress-plugin-framework/src/library/Core.php:60
9.	Pimple\Container->offsetGet()
io/plugins/publishpress/vendor/alledia/wordpress-plugin-framework/src/library/Core.php:102
10.	Allex\Container->Allex\{closure}()
io/plugins/publishpress/vendor/pimple/pimple/src/Pimple/Container.php:118
11.	Allex\M\Upgrade->__construct()
io/plugins/publishpress/vendor/alledia/wordpress-plugin-framework/src/library/Container.php:183
12.	Pimple\Container->offsetGet()
io/plugins/publishpress/vendor/alledia/wordpress-plugin-framework/src/library/Module/Upgrade.php:55
13.	Allex\Container->Allex\{closure}()
io/plugins/publishpress/vendor/pimple/pimple/src/Pimple/Container.php:118
14.	get_plugin_data()
io/plugins/publishpress/vendor/alledia/wordpress-plugin-framework/src/library/Container.php:104
15.	get_file_data()
wp-admin/includes/plugin.php:84
16.	fopen()
wp-includes/functions.php:4891
17.	wp-includes/functions.php:4891

> 
> Installer fails on site with custom WP_CONTENT_DIR
> Not known why double slash either?

**Comment by cconversion** [comment]
fixed by replacing:

`wp-content/plugins` with `WP_CONTENT_FOLDERNAME . '/plugins'`

or replacing more, still using standard wp-config CONSTANTS for dynamic read:
`ABSPATH . 'wp-content/plugins/` with `WP_CONTENT_DIR. '/'`

**Comment by stevejburge** [comment]
Solved in 1.12.0
